adapters/Unity/UnityAdapter/GADMAdapterUnityRewardedAd.m

```diff
@@ -102,91 +100,73 @@ - (void)presentFromViewController:(nonnull UIViewController *)viewController {
     GADMAdapterUnityMapTableRemoveObjectForKey(_placementInUseRewarded, self->_placementID);
   });
 
-  if (![UnityAds isReady:_placementID]) {
-    NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(
-        GADMAdapterUnityErrorShowAdNotReady, @"Failed to show Unity Ads rewarded video.");
-    [_adEventDelegate didFailToPresentWithError:error];
-    return;
-  }
   [_adEventDelegate willPresentFullScreenView];
-  [UnityAds addDelegate:self];
-  [UnityAds show:viewController placementId:_placementID];
+  [UnityAds show:viewController placementId:_placementID showDelegate:self];
 }
 
-#pragma mark - UnityAdsExtendedDelegate Methods
+- (void)cleanup {
+  // Clean up references to prevent a memory leak where the object can not be cleand up
```

Comment:
Which object is leaking? who is holding onto the reference? This class's NSMapTable has a weak reference to GADMAdapterUnityRewardedAd. Does the UnityAds SDK keep a strong reference to the delegate passed into [UnityAds load]? If not, then there shouldn't be any leaks.